RileyLinkKit/PumpOpsSynchronous.swift

```diff
@@ -510,18 +510,24 @@ class PumpOpsSynchronous {
                 timestamp.timeZone = pump.timeZone
 
                 if let date = timestamp.date?.addingTimeInterval(timeAdjustmentInterval) {
-                    if date.timeIntervalSince(startDate) < -eventTimestampDeltaAllowance {
-                        NSLog("Found event at (%@) to be more than %@s before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
-                        return (events: events, hasMoreEvents: false)
-                    } else if date.timeIntervalSince(timeCursor) > eventTimestampDeltaAllowance {
-                        NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
-                        return (events: events, hasMoreEvents: false)
-                    } else {
-                        if (date.compare(startDate) != .orderedAscending) {
-                            timeCursor = date
+                    
+                    let possibleCancelBecauseOfDate = !event.canBeDelayedAppend(withPumpModel: pumpModel)
```

Comment:
When we hit an event older than startTime, we're not 'canceling' the operation.  We've reached the end successfully.

Also, it's not "can" be delayed, as in a possibility.  When we have the pumpModel and eventType, we know with certainty if it is a delayed append event or not. 

And a delayed append event does not obviate the need to check for lost pump time.

And we can get rid of eventTimestampDeltaAllowance

```
if date <= startDate && !event.delayedAppend(withPumpModel: pumpModel) {
    // Success, we have all the events we need
    //NSLog("Found event at or before startDate(%@)", date as NSDate, String(describing: eventTimestampDeltaAllowance), startDate as NSDate);
    return (events: events, hasMoreEvents: false, cancelledEarly: false)
} else if date.timeIntervalSince(timeCursor) > TimeInterval(minutes: 60) {
    NSLog("Found event (%@) out of order in history. Ending history fetch.", date as NSDate)
    return (events: events, hasMoreEvents: false, cancelledEarly: true)
}
```